model/job.py

```diff
@@ -14,11 +14,20 @@ def cancel(self):
     def retrigger(self):
         self._execute_action(action='retrigger')
 
+    def log(self):
+        print(self._get_joblog())
+
     def _execute_action(self, action):
         rest_resource = 'jobs/{}/{}'.format(self.id, action)
         self.client.execute_request(http_verb='post', rest_resource=rest_resource)
         print('Job {} {} ({}) {}ed'.format(self.group_symbol, self.type_symbol, self.id, action))
 
+    def _get_joblog(self):
+        import requests
+        url = '{}/api/project/{}/job-log-url/?job_id={}'.format(self.client.TREEHERDER_BASE_URL, self.client.repo, self.id)
```

Comment:
My bad, I should have made `TreeherderClient.TREEHERDER_BASE_URL` and `TreeherderClient.repo` private. Let's use the request generation in `TreeherderClient` only. So `Job` won't have that responsability.  Here you can call it this way:
``` python
url = self.client.execute_request(rest_resource='job-log-url', get_arguments={job_id: self.id})
return self.client.execute_raw_request(url=url)
```
In TreeherderClient:
``` python
def execute_request(self, http_verb='get', rest_resource='', get_arguments=None):
    get_query = self._craft_get_query(get_arguments)
    url = '{}/api/project/{}/{}/{}'.format(self.TREEHERDER_BASE_URL, self.repo, rest_resource, get_query)
    return self.execute_raw_request(url, http_verb)

def execute_raw_request(url, http_verb='get'):
    headers = {'Referer': self.TREEHERDER_BASE_URL, 'X-CSRFToken': self.csrf_token}
    cookies = {'sessionid': self.session_id, 'csrftoken': self.csrf_token}

    http_method = getattr(requests, http_verb)
    response = http_method(url, headers=headers, cookies=cookies)
    response.raise_for_status()
    return response.json()
```

What do you think?